pynemo/profiler.py

```diff
@@ -388,31 +388,38 @@ def write_tidal_data(setup_var, dst_coord_var, grid, tide_cons, cons):
             
             nemo_bdy_tide_ncgen.CreateBDYTideNetcdfFile(fout_tide, 
                             val['nx'], 
-                            dst_coord_var.lonlat['t']['lon'].shape[1],
-                            dst_coord_var.lonlat['t']['lon'].shape[0], 
+                            dst_coord_var.lonlat[key]['lon'].shape[1],  # "key" is the grd value
+                            dst_coord_var.lonlat[key]['lon'].shape[0],
                             val['des']+tide_con, 
                             setup_var.settings['fv'], key.upper())
-            
-            ncpop.write_data_to_file(fout_tide, val['nam']+'1', 
-                                     cons['cos'][val['nam']][indx])
-            ncpop.write_data_to_file(fout_tide, val['nam']+'2', 
-                                     cons['sin'][val['nam']][indx])
-            ncpop.write_data_to_file(fout_tide, 'bdy_msk',
-                                     dst_coord_var.bdy_msk)
+
+            # Set the index for the con(stituent) to save
+            if val['nam'] == "z":
+                indx = 0
+                # bdy_msk is only defined on the t-grid (not u/v-points)
+                ncpop.write_data_to_file(fout_tide, 'bdy_msk',
+                                         dst_coord_var.bdy_msk)
+            elif val['nam'] == "u":
+                indx = 2
+            elif val['nam'] == "v":
+                indx = 4
+            else:
+                logging.error("profiler: Ooo, that should not have happened")
+
+            ncpop.write_data_to_file(fout_tide, val['nam']+'1',
+                                     cons[indx][int(tide_con)-1])  # "cos[var][constituent index]"
```

Comment:
the variables `cons` seems to have had a bit of a restructure, no longer with `cos` and `sin` subvariables. These are now references by indices.